app/reducers/modal.js

```diff
@@ -0,0 +1,81 @@
+// ------------------------------------
+// Initial State
+// ------------------------------------
+const initialState = {
+  modals: []
+}
+
+// ------------------------------------
+// Constants
+// ------------------------------------
+export const OPEN_MODAL = 'OPEN_MODAL'
+export const CLOSE_MODAL = 'CLOSE_MODAL'
+
+/**
+ * Generates uniq id for modals
+ */
+const genId = () =>
+  Math.random()
+    .toString(36)
+    .substring(7)
+
+// ------------------------------------
+// Actions
+// ------------------------------------
+export function openModal(type) {
+  const modal = {
+    id: genId(),
+    type
+  }
+  return {
+    type: OPEN_MODAL,
+    modal
+  }
+}
+
+export function closeModal(id) {
+  return {
+    type: CLOSE_MODAL,
+    id
+  }
+}
+
+// ------------------------------------
+// Action Handlers
+// ------------------------------------
+const ACTION_HANDLERS = {
+  [OPEN_MODAL]: (state, { modal }) => ({
+    ...state,
+    modals: [...state.modals, modal]
+  }),
+  [CLOSE_MODAL]: (state, { id }) => {
+    if (id) {
+      return {
+        ...state,
+        modals: state.modals.filter(item => item.id !== id)
+      }
+    }
+    return {
+      ...state,
+      modals: state.modals.filter((item, index) => index !== state.modals.length - 1)
```

Comment:
This looks like comparing an index to a constant (last modal index). Can we avoid using filter here and just use that last element ?

Reply:
Updated.